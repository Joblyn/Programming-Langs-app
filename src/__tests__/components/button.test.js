import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import Button, { OptButton, LargeButton } from "../../components/button/button";

describe("<Button />", () => {
  afterEach(cleanup);

  // test button redering
  it("renders the <Button /> with children", () => {
    const { container, getByText } = render(
      <Button href="/signup" outline>
        Sign up
      </Button>
    );

    expect(getByText("Sign up")).toBeTruthy();
    expect(container.firstChild.getAttribute("href")).toBe("/signup");
    expect(container.firstChild.matches(".bg-white.text-green"));
    expect(container.firstChild).toMatchSnapshot();
  });

  // test button rendering with "large" prop
  it("renders the large <Button />", () => {
    const { container, getByText } = render(
      <BrowserRouter>
        <Button href="/signup" outline large>
          Sign up
        </Button>
      </BrowserRouter>
    );

    expect(getByText("Sign up").matches(".w-52"));
    expect(container.firstChild).toMatchSnapshot();
  });

  //test Optbutton
  it("renders the <OptButton />", () => {
    const choices = ["Yes", "No", "Maybe"];
    let checked = "";
    const clickAction = (choice) => {
      checked = choice;
    };
    const { container, getByText, getByTestId } = render(
      <div>
        {choices.map((choice, id) => (
          <OptButton
            key={choice}
            value={choice}
            checked={checked}
            action={() => clickAction(choice)}
          />
        ))}
      </div>
    );
    expect(getByText("Yes")).toBeTruthy();
    expect(getByText("No")).toBeTruthy();
    expect(getByText("Maybe")).toBeTruthy();

    const parentElement = getByText("Yes").parentElement;
    fireEvent.click(parentElement);
    // let checkedIconExists;
    // for (let i = 0; i < parentElement.children.length; i++) {
    //   if (parentElement.children[i].matches(".checked-icon")) {
    //     checkedIconExists = true;
    //     return;
    //   }
    // }
    expect(checked === "Yes").toBeTruthy();
    // expect(getByTestId("CheckOutlinedIcon").parentElement === parentElement).toBeTruthy();
    // expect(checkedIconExists).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  // <LargeButton />
  it("renders the <LargeButton />", () => {
    const { container, getByText } = render(<LargeButton innerText={"Next"} />);

    expect(getByText("Next")).toBeTruthy();
    expect(
      getByText("Next").parentElement.matches(
        ".text-green.border.border-green.bg-white.cursor-pointer"
      )
    ).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the <LargeButton /> with type=next and disabled", () => {
    const { container, getByText } = render(
      <LargeButton type="next" disabled={true} innerText="Next" />
    );

    expect(getByText("Next")).toBeTruthy();
    expect(
      getByText("Next").parentElement.matches(
        ".text-white.bg-green.cursor-not-allowed.bg-opacity-50"
      )
    ).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
