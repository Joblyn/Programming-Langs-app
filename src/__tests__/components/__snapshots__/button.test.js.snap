// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders the <Button /> with children 1`] = `
<a
  class="btn rounded-md px-1.5 md:px-2 py-1 md:py-1.5 mx-1 md:mx-2 bg-white text-green border border-green text-sm md:text-base hover:bg-lightgreen hover:text-green cursor-pointer transition-all inline-flex items-center justify-center select-none text-center md:h-10 w-20 shadow-md"
  href="/signup"
>
  Sign up
</a>
`;

exports[`<Button /> renders the <LargeButton /> 1`] = `
<div
  class="w-full lg:max-w-sm text-green border border-green bg-white active:scale-95 uppercase cursor-pointer hover:opacity-80  p-3 text-2xl flex justify-center items-center rounded-md my-3 btn"
>
  <span
    class="px-[7px] font-bold tracking-wider"
  >
    Next
  </span>
</div>
`;

exports[`<Button /> renders the <LargeButton /> with type=next and disabled 1`] = `
<div
  class="w-full lg:max-w-sm text-white bg-green active:scale-95 uppercase cursor-not-allowed bg-opacity-50  p-3 text-2xl flex justify-center items-center rounded-md my-3 btn"
>
  <span
    class="px-[7px] font-bold tracking-wider"
  >
    Next
  </span>
</div>
`;

exports[`<Button /> renders the <OptButton /> 1`] = `
<div>
  <div
    class="w-full lg:max-w-sm rounded-md bg-white border border-green text-lg flex justify-center items-center my-2 btn hover:bg-green hover:text-white text-black-primaryrelative opt-cont  opt-cont active:scale-95"
  >
    <label
      class="w-full p-2.5 opt-label cursor-pointer relative"
    >
      <span
        class="tracking-wider w-full text-center"
      >
        Yes
      </span>
    </label>
  </div>
  <div
    class="w-full lg:max-w-sm rounded-md bg-white border border-green text-lg flex justify-center items-center my-2 btn hover:bg-green hover:text-white text-black-primaryrelative opt-cont  opt-cont active:scale-95"
  >
    <label
      class="w-full p-2.5 opt-label cursor-pointer relative"
    >
      <span
        class="tracking-wider w-full text-center"
      >
        No
      </span>
    </label>
  </div>
  <div
    class="w-full lg:max-w-sm rounded-md bg-white border border-green text-lg flex justify-center items-center my-2 btn hover:bg-green hover:text-white text-black-primaryrelative opt-cont  opt-cont active:scale-95"
  >
    <label
      class="w-full p-2.5 opt-label cursor-pointer relative"
    >
      <span
        class="tracking-wider w-full text-center"
      >
        Maybe
      </span>
    </label>
  </div>
</div>
`;

exports[`<Button /> renders the large <Button /> 1`] = `
<a
  class="btn rounded-md p-3 mx-1 md:mx-2 bg-white text-green border border-green cursor-pointer transition-all hover:bg-lightgreen hover:text-green inline-flex items-center justify-center select-none text-center shadow-md md:w-80 w-52 uppercase text-2xl animation_click"
  href="/signup"
>
  Sign up
</a>
`;
